tests/unit/test_app_plugins.py

```diff
@@ -0,0 +1,227 @@
+# This file is part of craft_application.
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for craft-application app plugins."""
+
+import argparse
+import os
+import sys
+import textwrap
+from importlib.metadata import Distribution, DistributionFinder
+from typing import cast
+
+import pytest
+from craft_cli import emit
+from overrides import override
+
+from craft_application import (
+    Application,
+    AppService,
+    CraftApplicationPluginProtocol,
+    commands,
+    services,
+)
+
+FAKE_APP = "boopcraft"
+PLUGIN_GROUP_NAME = "craft_application_plugins.application"
+PLUGIN_ENTRY_POINT_NAME = f"{FAKE_APP}_plugin"
+PLUGIN_MODULE_NAME = f"{FAKE_APP}_module"
+
+
+class FakeCraftApplicationPlugin(CraftApplicationPluginProtocol):
+    def configure(self, app: Application) -> None:
+        assert app.services.fake
```

Comment:
I'm not sure if accessing by name is a recommended solution after Alex added `.register()` capabilities, but I may be wrong (I think it's there just for backward compatibility).
```suggestion
        assert app.services.get("fake"), "Fake service is not provided by app."
```